source/gfoidl.Stochastics/Statistics/Sample.AverageVarianceCore.cs

```diff
@@ -99,8 +98,8 @@ internal unsafe void CalculateAverageAndVarianceCoreImpl(int idxStart, int idxEn
                     var var3 = Vector<double>.Zero;
 
                     // https://github.com/gfoidl/Stochastics/issues/46
-                    int m = n & ~(8 * Vector<double>.Count - 1);
-                    for (; i < m; i += 8 * Vector<double>.Count)
+                    workEnd = start + (n & ~(8 * Vector<double>.Count - 1));
```

Comment:
Code isn't so obvious to read. Debugging gets really bad (pointer-addresses vs. integer-values).

Base version:
```asm
       C4417D1091A0000000   vmovupd  ymm10, ymmword ptr[r9+160]
       C4C16558DA           vaddpd   ymm3, ymm10
       C4412D59D2           vmulpd   ymm10, ymm10
       C4C14558FA           vaddpd   ymm7, ymm10
       C4417D1091C0000000   vmovupd  ymm10, ymmword ptr[r9+192]
       C4C15D58E2           vaddpd   ymm4, ymm10
       C4412D59D2           vmulpd   ymm10, ymm10
       C4413D58C2           vaddpd   ymm8, ymm10
       C4417D1091E0000000   vmovupd  ymm10, ymmword ptr[r9+224]
       C4C15558EA           vaddpd   ymm5, ymm10
       C4412D59D2           vmulpd   ymm10, ymm10
       C4413558CA           vaddpd   ymm9, ymm10
       4981C100010000       add      r9, 256
       83C620               add      esi, 32
       413BF2               cmp      esi, r10d
       0F8C3AFFFFFF         jl       G_M16821_IG10

G_M16821_IG11:
       448BD2               mov      r10d, edx
       4183E2F0             and      r10d, -16
       413BF2               cmp      esi, r10d
       7D5D                 jge      SHORT G_M16821_IG12
       C4417D1011           vmovupd  ymm10, ymmword ptr[r9]
       C4C16D58D2           vaddpd   ymm2, ymm10
       C4412D59D2           vmulpd   ymm10, ymm10
       C4C14D58F2           vaddpd   ymm6, ymm10
       C4417D105120         vmovupd  ymm10, ymmword ptr[r9+32]
       C4C16558DA           vaddpd   ymm3, ymm10
```

Variant:
```asm
       C4417D1091A0000000   vmovupd  ymm10, ymmword ptr[r9+160]
       C4C16558DA           vaddpd   ymm3, ymm10
       C4412D59D2           vmulpd   ymm10, ymm10
       C4C14558FA           vaddpd   ymm7, ymm10
       C4417D1091C0000000   vmovupd  ymm10, ymmword ptr[r9+192]
       C4C15D58E2           vaddpd   ymm4, ymm10
       C4412D59D2           vmulpd   ymm10, ymm10
       C4413D58C2           vaddpd   ymm8, ymm10
       C4417D1091E0000000   vmovupd  ymm10, ymmword ptr[r9+224]
       C4C15558EA           vaddpd   ymm5, ymm10
       C4412D59D2           vmulpd   ymm10, ymm10
       C4413558CA           vaddpd   ymm9, ymm10
       4981C100010000       add      r9, 256
       4C3BCE               cmp      r9, rsi
       0F823DFFFFFF         jb       G_M16821_IG10

G_M16821_IG11:
       8BF2                 mov      esi, edx
       83E6F0               and      esi, -16
       4863F6               movsxd   rsi, esi
       488D34F7             lea      rsi, [rdi+8*rsi]
       4C3BCE               cmp      r9, rsi
       735A                 jae      SHORT G_M16821_IG12
       C4417D1011           vmovupd  ymm10, ymmword ptr[r9]
       C4C16D58D2           vaddpd   ymm2, ymm10
       C4412D59D2           vmulpd   ymm10, ymm10
       C4C14D58F2           vaddpd   ymm6, ymm10
       C4417D105120         vmovupd  ymm10, ymmword ptr[r9+32]
       C4C16558DA           vaddpd   ymm3, ymm10
```

In the base version the `add`s are done in parallel (ISP), so no real improvement.
Code below label `G_M16821_IG11` is more, so a disadvantage.

In summary it's nice that it's possible and all tests pass, but the disadvantages overwigth. So revert this commit.